centreon/www/front_src/src/ResourceAccessManagement/AddEditResourceAccessRule/FormInputs/components/DatasetFilters.tsx

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable react/no-array-index-key */
+import { ReactElement } from 'react';
+
+import { equals, isEmpty, isNil, last } from 'ramda';
+
+import useDatasetFilters from '../hooks/useDatasetFilters';
+import { Dataset } from '../../../models';
+import { useDatasetFiltersStyles } from '../styles/DatasetFilters.styles';
+
+import DatasetFilter from './DatasetFilter';
+import AddDatasetButton from './AddDatasetButton';
+import DeleteDatasetButton from './DeleteDatasetButton';
+import DatasetFilterDivider from './DatasetFilterDivider';
+
+const DatasetFilters = (): ReactElement => {
+  const { classes } = useDatasetFiltersStyles();
+  const { addDatasetFilter, datasetFilters, deleteDatasetFilter } =
+    useDatasetFilters();
+
+  const areResourcesFilled = (datasets: Array<Dataset>): boolean =>
+    datasets?.every(
+      ({ resourceType, resources }) =>
+        !isEmpty(resourceType) && !isEmpty(resources)
+    );
+
+  const areAddDatasetButtonDisabled = (
+    datasetFiltersArray: Array<Array<Dataset>>
+  ): boolean => {
+    if (isNil(datasetFiltersArray)) {
+      return false;
+    }
+
+    const lastDatasetFilter = last(datasetFiltersArray);
+    if (isNil(lastDatasetFilter)) {
+      return false;
+    }
+
+    const lastDataset = last(lastDatasetFilter);
+    if (isNil(lastDataset)) {
+      return false;
+    }
+
+    return isEmpty(lastDataset.resourceType) || isEmpty(lastDataset.resources);
+  };
+
```

Comment:
this logic can be simplified. for example you can combine the three "if condition" into a single one
`    const lastDataset = last(lastDatasetFilter);

    if (isNil(datasetFiltersArray) || isNil(lastDataset) || isNil( last(lastDataset)) ) {
        return false;
    }`
You can further simplify the code based on the specific logic you intend to implement.

Reply:
is datasetFiltersArray can be null ?